src/sliceDataStorage.cpp

```diff
@@ -115,6 +115,10 @@ bool SliceMeshStorage::getExtruderIsUsed(int extruder_nr) const
     {
         return true;
     }
+    if ((getSettingAsCount("top_layers") > 0 || getSettingAsCount("bottom_layers") > 0) && getSettingAsCount("roofing_layer_count") > 0 && getSettingAsExtruderNr("roofing_extruder_nr") == extruder_nr)
```

Comment:
It might be an idea to nest this together with the previous `if` statement. Sort of subjective style though.

(Not blocking.)